x/blob/types/blob_tx.go

```diff
@@ -0,0 +1,118 @@
+package types
+
+import (
+	"bytes"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/nmt/namespace"
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+// ProcessedBlobTx caches the unmarshalled result of the BlobTx
+type ProcessedBlobTx struct {
+	Blobs []tmproto.Blob
+	Tx    []byte
+}
+
+// NewBlob creates a new coretypes.Blob from the provided data after performing
+// basic stateless checks over it.
+func NewBlob(ns namespace.ID, blob []byte) (*tmproto.Blob, error) {
+	err := ValidateBlobNamespaceID(ns)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(blob) == 0 {
+		return nil, ErrZeroBlobSize
+	}
+
+	return &tmproto.Blob{
+		NamespaceId:  ns,
+		Data:         blob,
+		ShareVersion: uint32(appconsts.ShareVersionZero),
+	}, nil
+}
+
+// ProcessBlobTx performs stateless checks on the BlobTx to ensure that the
+// blobs attached to the transaction are valid. During this process, it
+// separates the blobs from the MsgPayForBlob, which are returned in the
+// ProcessedBlobTx.
+//
+// NOTE: ProcessBlobTx does not call the ValidateBasic method on either the
+// transaction or messages in that transaction.
+func ProcessBlobTx(txcfg client.TxEncodingConfig, bTx tmproto.BlobTx) (ProcessedBlobTx, error) {
+	sdkTx, err := txcfg.TxDecoder()(bTx.Tx)
+	if err != nil {
+		return ProcessedBlobTx{}, err
+	}
```

Comment:
[for reviewers]
this function verifies that the blob transaction is valid. Notably, it performs important check that otherwise would have been performed in the `ValidateBasic` method of `MsgWirePFB`. It's very important that we call this function during check tx and prepare proposal